pypesto/profile/profile_next_guess.py

```diff
@@ -0,0 +1,280 @@
+import numpy as np
+import copy
+
+
+def next_guess(x,
+               par_index,
+               par_direction,
+               profile_options,
+               update_type,
+               current_profile,
+               problem,
+               global_opt):
+    """
+       This is function initializes profiling based on a previous optimization.
+
+       Parameters
+       ----------
+
+        x: ndarray
+           The current position of the profiler
+
+        par_index: ndarray
+            The index of the current profile
+
+        par_direction: int
+            The direction, in which the profiling is done (1 or -1)
+
+        profile_options: pypesto.ProfileOptions, optional
+            Various options applied to the profile optimization.
+
+        update_type: basestring
+            type of update for next profile point
+
+        current_profile: pypesto.ProfilerResults
+            The profile which should be computed
+
+        problem: pypesto.Problem
+            The problem to be solved.
+
+        global_opt: float
+            log-posterior value of the global optimum
+       """
+
+    if update_type == 'fixed_step':
+        return fixed_step(x, par_index, par_direction, profile_options,
+                          problem)
+    else:
+        if update_type == 'adaptive_step_order_0':
+            order = 0
+        elif update_type == 'adaptive_step_order_1':
+            order = 1
+        elif update_type == 'adaptive_step_regression':
+            order = float('nan')
+        else:
+            raise Exception('Unsupported update_type for '
+                            'create_next_startpoint.')
+
+        return adaptive_step(x, par_index, par_direction, profile_options,
+                             current_profile, problem, global_opt, order)
+
+
+def fixed_step(x, par_index, par_direction, options, problem):
+    delta_x = np.zeros(len(x))
+    delta_x[par_index] = par_direction * options.default_step_size
+
+    # check whether the next point is maybe outside the bounds
+    # and correct it
+    next_x_par = x[par_index] + delta_x[par_index]
+    if par_direction is -1:
+        if next_x_par < problem.lb_full[par_index]:
+            delta_x[par_index] = problem.lb_full[par_index] - x[par_index]
+    else:
+        if next_x_par > problem.ub_full[par_index]:
+            delta_x[par_index] = problem.ub_full[par_index] - x[par_index]
+
+    return x + delta_x
+
+
+def adaptive_step(x, par_index, par_direction, options, current_profile,
+                  problem, global_opt, order=1):
+    # There is this magic factor in the old profiling code which slows down
+    # profiling at small ratios (must be >= 0 and < 1)
```

Comment:
wow, this method is quite long. it looks well-written (and documented!), but would it be possible to decompose? maybe the nested functions?